src/Core/Rendering/RenderController.cpp

```diff
@@ -537,6 +538,62 @@ namespace MxEngine
         this->RenderToTexture(output, shader);
     }
 
+    void RenderController::ApplyGodRayEffect(CameraUnit& camera, TextureHandle& input, TextureHandle& output)
+    {
+        if (!camera.Effects->IsGodRayEnabled())
+            return;
+        int dirLightCount = this->Pipeline.Lighting.DirectionalLights.size();
+        if (dirLightCount <= 0)
+            return;
+        auto godRayShader = this->Pipeline.Environment.Shaders["GodRay"_id];
+        godRayShader->Bind();
+        godRayShader->IgnoreNonExistingUniform("camera.viewProjMatrix");
+        godRayShader->IgnoreNonExistingUniform("normalTex");
+        godRayShader->IgnoreNonExistingUniform("albedoTex");
+        godRayShader->IgnoreNonExistingUniform("materialTex");
+
+        Texture::TextureBindId textureId = 0;
+        this->BindGBuffer(camera, *godRayShader, textureId);
+        this->BindCameraInformation(camera, *godRayShader);
+
+        input->Bind(textureId++);
+        godRayShader->SetUniform("cameraOutput", input->GetBoundId());
+
+        {
+            constexpr size_t MaxDirLightCount = 4;
```

Comment:
its already declared in upper scope (https://github.com/asc-community/MxEngine/blob/b82dbeefac416c15d44bf134058f668eb071078e/src/Core/Rendering/RenderController.cpp#L49)

Reply:
also see its redaclared by me in directional light code, thats mine mistake too
https://github.com/asc-community/MxEngine/blob/b82dbeefac416c15d44bf134058f668eb071078e/src/Core/Rendering/RenderController.cpp#L425